File: proyecto de ciclo/proyectoCicloPython/matrices.py
print('Programa para sacar el promedio de numero menores en cada ila de una matriz y ver si es igual'
      ' a alguna dato en las esquinas de otra matriz')
prom=0; sum=0; menor=0;

A = []
B = []
print("Ingresar los elementos de la matriz A")
for i in range(0, 5):
    A.append([])
    for j in range(0, 5):
        pos = int(input("A[" + str(i) + "]["+str(j)+"]= "))
        A[i].append(pos)

print('\n')
print("Ingresar los elementos de la matriz B")
for i in range(0, 5):
    B.append([])
    for j in range(0, 5):
        pos = int(input("B[" + str(i) + "]["+str(j)+"]= "))
        B[i].append(pos)
print('--------------------------------------------------------')
print('Matriz A')
for i in range(0, 5):
    for j in range(0, 5):
        print(A[i][j], end='   ')
    print('  ')

print('\n')
print('------------------------------------------------------')
print('Matriz B')
for i in range(0, 5):
    for j in range(0, 5):
        print(B[i][j], end='   ')
    print('  ')

print('------------------------------------------------------')

print('Menores en matriz A')
for i in range(0, 5):
    menor = A[i][0]
    for j in range(0, 5):
        if menor > A[i][j]:
            menor = A[i][j]
    sum = sum + menor

print("SUMA DE MENORES "+ str(sum))
prom = int(sum/5)
print('El promedio de los menores de las filas de la matriz A es: '+ str(prom))
if prom == B[0][0] :
    print("El promedio de las filas de la matriz A es igual a la esquina en la poscición B[0][0]")

if(prom == B[0][4] ):
    print("El promedio de las filas de la matriz A es igual a la esquina en la poscición B[0][4]")

if(prom == B[4][0] ):
    print("El promedio de las filas de la matriz A es igual a la esquina en la poscición B[4][0]")

if(prom == B[4][4] ):
    print("El promedio de las filas de la matriz A es igual a la esquina en la poscición B[4][4]")

print('------------------------------------------------------')
sum = 0
print('Menores en matriz B')
for i in range(0, 5):
    menor = B[i][0]
    for j in range(0, 5):
        if menor > B[i][j]:
            menor = B[i][j]
    sum = sum + menor

print("SUMA DE MENORES "+ str(sum))
prom = int(sum/5)
print('El promedio de los menores de las filas de la matriz A es: '+ str(prom))
if prom == A[0][0] :
    print("El promedio de las filas de la matriz A es igual a la esquina en la poscición A[0][0]")

if(prom == A[0][4] ):
    print("El promedio de las filas de la matriz A es igual a la esquina en la poscición A[0][4]")

if(prom == A[4][0] ):
    print("El promedio de las filas de la matriz A es igual a la esquina en la poscición A[4][0]")

if(prom == A[4][4] ):
    print("El promedio de las filas de la matriz A es igual a la esquina en la poscición A[4][4]")

File: proyecto de ciclo/proyectoCiclo/src/proyectociclo/matrices.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package proyectociclo;

import java.util.Scanner;

/**
 *
 * @author kevin
 */
public class matrices {
    public static void main(String[] args) {
        /*Leer dos matrices 5x5 enteras y determinar 
        si el promedio entero de los números 
        menores por cada fila de una matriz 
        corresponde a alguno de los datos 
        almacenados en las “esquinas” de la otra 
        matriz.*/
        Scanner leer = new Scanner(System.in);
        int menor,suma,i,j, promedio;
        menor=0;suma=0;promedio=0;
        
        
        int[][] A = new int [5][5];
        int[][] B = new int [5][5];      
        // Ingresar vector A
        System.out.println("\nINGRESAR DATOS DEL MATRIZ A");
        for(i=0;i<5;i++){
            for(j=0;j<5;j++){
                System.out.print("A ["+i+"] ["+j+"] = ");
                A[i][j] = leer.nextInt();
            }
        }
        //Ingresar vector B
        System.out.println("INGRESAR DATOS DEL MATRIZ B");
        for(i=0;i<5;i++){
            for(j=0;j<5;j++){
                System.out.print("B ["+i+"] ["+j+"] = ");
                B[i][j] = leer.nextInt();
            }
        }
        
        System.out.println("MATRIZ A");
        //presentar vector A
        for(i=0; i<5; i++) {
            for(j=0; j<5 ; j++) {
                System.out.print(A[i][j] + "  ");
            }
            System.out.println(" ");
        }
        System.out.println("MATRIZ B");
        // Presentar vector B
        for(i = 0; i < 5; i++) {
            for(j = 0;j < 5 ; j++) {
                System.out.print(B[i][j] + "  ");
            }
            System.out.println(" ");
        }
        
        System.out.println("--------------------------------");
        System.out.println("Menores en matriz A");
        // DETECTAR EL MENOR DE CADA FILA EN MATRIZ A
        for (i = 0; i < 5; i++) {
            menor = A[i][0];
            for(j = 0;j < 5 ; j++) {
                if(menor > A[i][j] ){
                    menor = A[i][j];
                }
            }
            suma = suma + menor;
        }
        System.out.println("SUMA DE MENORES "+suma);
        
        promedio = (suma/5);
        System.out.println(promedio);
        
        if(promedio == B[0][0] ){
            System.out.println("El promedio de las filas de la matriz A"
                    + " es igual a la esquina en la poscición B[0][0]");
        }
        if(promedio == B[0][4] ){
            System.out.println("El promedio de las filas de la matriz A"
                    + " es igual a la esquina en la poscición B[0][4]");
        }
        if(promedio == B[4][0] ){
            System.out.println("El promedio de las filas de la matriz A"
                    + " es igual a la esquina en la poscición B[4][0]");
        }
        if(promedio == B[4][4] ){
            System.out.println("El promedio de las filas de la matriz A"
                    + " es igual a la esquina en la poscición B[4][4]");
        }
            
        System.out.println("--------------------------------");
        suma=0;
        System.out.println("Menores en matriz B");
        // DETECTAR EL MENOR DE CADA FILA EN MATRIZ A
        for (i = 0; i < 5; i++) {
            menor = B[i][0];
            for(j = 0;j < 5 ; j++) {
                if(menor > B[i][j] ){
                    menor = B[i][j];
                }
            }
            suma = suma + menor;
        }
        System.out.println("SUMA DE MENORES "+suma);
        
        promedio = (suma/5);
        System.out.println(promedio);
        
        if(promedio == A[0][0] ){
            System.out.println("El promedio de las filas de la matriz B"
                    + " es igual a la esquina en la poscición A[0][0]");
        }
        if(promedio == A[0][4] ){
            System.out.println("El promedio de las filas de la matriz B"
                    + " es igual a la esquina en la poscición A[0][4]");
        }
        if(promedio == A[4][0] ){
            System.out.println("El promedio de las filas de la matriz B"
                    + " es igual a la esquina en la poscición A[4][0]");
        }
        if(promedio == A[4][4] ){
            System.out.println("El promedio de las filas de la matriz B"
                    + " es igual a la esquina en la poscición A[4][4]");
        }
            
    }
}
